tests/sandbox/EqualityComparable.h

```diff
@@ -0,0 +1,76 @@
+// FIXME remove pragma after moving to cts
+#pragma once
+
+#include "../common/common.h"
+
+#include "TypeTraits.h"
+
+#include <chrono>
+#include <iostream>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+template <typename It>
+constexpr bool check_equality_comparable_requirement(It valid_iterator) {
+  // static_assert(type_traits::has_equality_operator_v<It>);
+  
+  // Expect to fail a compilation if equal operator is not implemented
+  static_cast<void>((valid_iterator == valid_iterator));
+
+  It a = valid_iterator;
+  It b = valid_iterator;
+  It c = valid_iterator;
+  const It const_a = valid_iterator;
+  const It const_b = valid_iterator;
+  const It const_c = valid_iterator;
+
+  bool result = false;
+
+  {
+    bool a_eq_b = a == b;
+    bool b_eq_a = b == a;
+    bool b_eq_c = b == c;
+    bool a_eq_c = a == c;
+
+    result = a_eq_b;
+    result &= b_eq_a;
+    result &= b_eq_c;
+    result &= a_eq_c;
+
+    bool returns_bool = false;
+    returns_bool = std::is_same_v<decltype((a == b)), bool>;
+    returns_bool &= std::is_same_v<decltype((b == a)), bool>;
+    returns_bool &= std::is_same_v<decltype((b == c)), bool>;
+    returns_bool &= std::is_same_v<decltype((a == c)), bool>;
+
+    result &= returns_bool;
```

Comment:
By collecting results you are effectively hiding what exact expression caused Catch2 assertion failure.
We are based on Catch2 now, so why not use `STATIC_CHECK` from https://github.com/catchorg/Catch2/blob/devel/docs/other-macros.md?

Reply:
As STATIC_CHECK can fail a compilation, I decided to change it to CHECK